src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/ProductOfConditions.hpp

```diff
@@ -0,0 +1,474 @@
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace grmhd::GhValenciaDivClean::BoundaryConditions {
+namespace detail {
+
+// Implementation for expanding the combination of packs for full compatibility
+// with all possible combination of tag lists that can be used for constructing
+// `BoundaryCondition` structs.
+template <
+    typename DerivedGhCondition, typename DerivedValenciaCondition,
+    typename GhEvolvedTagList, typename ValenciaEvolvedTagList,
+    typename GhFluxTagList, typename ValenicaFluxTagList,
+    typename GhInteriorEvolvedTagList, typename ValenciaInteriorEvolvedTagList,
+    typename GhInteriorPrimitiveTagList, typename ValenciaInteriorPrimitiveTgs,
+    typename GhInteriorTempTagList, typename ValenciaInteriorTempTagList,
+    typename DeduplicatedTempTagList, typename GhInteriorDtTagList,
+    typename ValenciaInteriorDtTagList, typename GhInteriorDerivTagList,
+    typename ValenciaInteriorDerivTagList, typename GhGridlessTagList,
+    typename ValenciaGridlessTagList, typename DeduplicatedGridlessTagList>
+struct ProductOfConditionsImpl;
+
+template <
+    typename DerivedGhCondition, typename DerivedValenciaCondition,
+    typename... GhEvolvedTags, typename... ValenciaEvolvedTags,
+    typename... GhFluxTags, typename... ValenciaFluxTags,
+    typename... GhInteriorEvolvedTags, typename... ValenciaInteriorEvolvedTags,
+    typename... GhInteriorPrimitiveTags,
+    typename... ValenciaInteriorPrimitiveTags, typename... GhInteriorTempTags,
+    typename... ValenciaInteriorTempTags, typename... DeduplicatedTempTags,
+    typename... GhInteriorDtTags, typename... ValenciaInteriorDtTags,
+    typename... GhInteriorDerivTags, typename... ValenciaInteriorDerivTags,
+    typename... GhGridlessTags, typename... ValenciaGridlessTags,
+    typename... DeduplicatedGridlessTags>
+struct ProductOfConditionsImpl<
+    DerivedGhCondition, DerivedValenciaCondition, tmpl::list<GhEvolvedTags...>,
+    tmpl::list<ValenciaEvolvedTags...>, tmpl::list<GhFluxTags...>,
+    tmpl::list<ValenciaFluxTags...>, tmpl::list<GhInteriorEvolvedTags...>,
+    tmpl::list<ValenciaInteriorEvolvedTags...>,
+    tmpl::list<GhInteriorPrimitiveTags...>,
+    tmpl::list<ValenciaInteriorPrimitiveTags...>,
+    tmpl::list<GhInteriorTempTags...>, tmpl::list<ValenciaInteriorTempTags...>,
+    tmpl::list<DeduplicatedTempTags...>, tmpl::list<GhInteriorDtTags...>,
+    tmpl::list<ValenciaInteriorDtTags...>, tmpl::list<GhInteriorDerivTags...>,
+    tmpl::list<ValenciaInteriorDerivTags...>, tmpl::list<GhGridlessTags...>,
+    tmpl::list<ValenciaGridlessTags...>,
+    tmpl::list<DeduplicatedGridlessTags...>> {
+  // In the current setup, we aren't given type information about the possible
```

Comment:
I think the below will cover our immediate needs, but we probably will soon need support for `dg_ghost` on the GRMHD variables and `dg_time_derivative` on the GH. We don't have Bjorhus BCs for the GRMHD system, while we would want those for the GH system at the outer boundary. I don't know immediately how easily that can be handled generically. Now, as long as the outer boundary is away from the matter we can probably ignore this and hack in "zero time derivative"-ish BCs and let the variable fixing deal with any issues. we will have to see

Reply:
ah.. right. yikes, I think that will need changes to the `Impl` functions in the DG boundary condition routines, because they currently assume a single kind of boundary condition calculation I think.
Perhaps we can add a `tmpl::list` here for the set of evolved variables that use the `dg_ghost` scheme and one for the set that use `dg_time_derivative`, then the `Impl` function can loop through the possibilities, applying the appropriate conditions for the respective subsets of tags?

Reply:
yea, that sounds like it should work fine. We'll have to experiment once we actually have the evolutions going, so I wouldn't worry about it right now :)

Reply:
Okay, sounds good -- I'll start trying things for that once the Bjorhus BCs for GH are all in place.